tests/unit_tests/test_wave_06.py

```diff
@@ -76,16 +76,16 @@ def test_swap_best_by_category():
         their_priority="Decor"
     )
 
-    raise Exception("Complete this test according to comments below.")
-    # *********************************************************************
-    # ****** Complete Assert Portion of this test **********
-    # *********************************************************************
-    # Assertions should check:
-    # - That the results is truthy
-    # - That tai and jesse's inventories are the correct length
-    # - That all the correct items are in tai and jesse's inventories, including the items which were swapped from one vendor to the other
+    # Assert
+    assert result == True
+    assert len(tai.inventory) == 3
+    assert len(jesse.inventory) == 3
+    assert item_a and item_b and item_f in tai.inventory
+    assert item_d and item_e and item_c in jesse.inventory
```

Comment:
These checks are not working how you might expect. Based on python's operator precedence, `item_f in tai.inventory` will be evaluated first, which will evaluate to `True` then `item_d and item_e and True` will be evaluated. Since the Items are truthy, the expression will evaluate to true without confirming what we intend.

If we want to condense the checks for all items to a single line, one approach would be to convert the lists to sets and use set operations:
```python
assert {item_f, item_a, item_b}.issubset(tai.inventory)
assert {item_c, item_d, item_e}.issubset(jesse.inventory)
```
Another approach would be to use the `all` function:
```python
assert all(item in tai.inventory for item in [item_f, item_a, item_b])
assert all(item in jesse.inventory for item in [item_c, item_d, item_e])
```

Reply:
Oh wow, this is so helpful to know! I really like the second approach you list here. It's timely that you mention this, because I was working on a similar piece of logic in another coding project and had some funny stuff happening with my logic that I couldn't quite figure out. Thank you!